packages/server/test/regression/issue-949-inferProcedureInput.test.ts

```diff
@@ -0,0 +1,71 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+// IMPORTANT:
+// needs to be imported from compiled output otherwise we get a false-positive
+import * as trpc from '../../dist/trpc-server.cjs';
+import { expectTypeOf } from 'expect-type';
+import { z } from 'zod';
+
+// https://github.com/trpc/trpc/issues/949
+// https://github.com/trpc/trpc/pull/955
+test('inferProcedureFromInput regression', async () => {
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  type Context = {};
```

Comment:
Personal opinion: `{}` is fine and I usually have a good reason when I want to use it. I usually disable ban-types on repos where most people committing code are pretty familiar with typescript.